geode-core/src/distributedTest/java/org/apache/geode/cache/ConnectionPoolDUnitTest.java

```diff
@@ -3279,124 +2316,79 @@ public void run2() throws CacheException {
+      Region<Object, Object> region1 = getRootRegion().getSubregion(regionName1);
+      for (int i = 0; i < 20; i++) {
+        region1.put("key-string-" + i, "value-" + i);
       }
     });
 
     // Put some values on the client.
-    client.invoke("Put values client", new CacheSerializableRunnable() {
-      @Override
-      public void run2() throws CacheException {
-        Region<Object, Object> region1 = getRootRegion().getSubregion(regionName1);
+    client.invoke("Put values client", () -> {
+      Region<Object, Object> region1 = getRootRegion().getSubregion(regionName1);
 
-        for (int i = 0; i < 10; i++) {
-          region1.put("key-string-" + i, "client-value-" + i);
-        }
+      for (int i = 0; i < 10; i++) {
+        region1.put("key-string-" + i, "client-value-" + i);
       }
     });
 
-    SerializableRunnable closePool = new CacheSerializableRunnable("Close Pool") {
-      @Override
-      public void run2() throws CacheException {
-        Region<Object, Object> region1 = getRootRegion().getSubregion(regionName1);
-        String pName = region1.getAttributes().getPoolName();
-        region1.localDestroyRegion();
-        PoolImpl p = (PoolImpl) PoolManager.find(pName);
-        p.destroy();
-      }
-    };
-
-    client.invoke(closePool);
+    client.invoke("Close Pool", () -> {
+      Region<Object, Object> region1 = getRootRegion().getSubregion(regionName1);
+      String pName = region1.getAttributes().getPoolName();
+      region1.localDestroyRegion();
+      PoolImpl p = (PoolImpl) PoolManager.find(pName);
+      p.destroy();
+    });
 
-    SerializableRunnable validateClientRegisterUnRegister =
-        new CacheSerializableRunnable("validate Client Register UnRegister") {
-          @Override
-          public void run2() throws CacheException {
-            for (CacheServer cacheServer : getCache().getCacheServers()) {
-              CacheServerImpl bsi = (CacheServerImpl) cacheServer;
-              final CacheClientNotifierStats ccnStats =
-                  bsi.getAcceptor().getCacheClientNotifier().getStats();
-              WaitCriterion ev = new WaitCriterion() {
-                @Override
-                public boolean done() {
-                  return ccnStats.getClientRegisterRequests() == ccnStats
-                      .getClientUnRegisterRequests();
-                }
-
-                @Override
-                public String description() {
-                  return null;
-                }
-              };
-              await().untilAsserted(ev);
-              assertEquals("HealthMonitor Client Register/UnRegister mismatch.",
-                  ccnStats.getClientRegisterRequests(), ccnStats.getClientUnRegisterRequests());
-            }
-          }
-        };
+    server1.invoke("validate Client Register UnRegister", () -> {
+      for (CacheServer cacheServer : getCache().getCacheServers()) {
+        CacheServerImpl bsi = (CacheServerImpl) cacheServer;
```

Comment:
`InternalCacheServer` (if the methods exist on the interface or you can add the methods to it)